### ISSUE TITLE
About loss and auxiliary inputs

### ISSUE THREAD

**Issue by Shexiaox** [facebookresearch/vggt-omega#42]
Hi, excellent work, I’ve learnt a great deal.
I would like to ask about the reference to "we found that the aleatoric uncertainty (confidence) loss can be unstable during fine-tuning. " in the paper. 
Does this confidence loss refer to "loss_conf_depth" in VGGT?

Additionally, regarding auxiliary inputs, I would like to ask about the extra inputs used to constrain the camera pose. Do you think it is better to use them as token inputs before the aggregator, or as constraints on the camera head?
If a rough camera position is known, could it be converted into camera extrinsic parameters and used as an iterative prior for the camera head?
Regarding the temporal order, would a simple frame index be preferable, or a more complex encoding?

If one wishes to achieve a scale closer to reality in a single-class scenario, whilst maintaining data normalisation during training and training a metric-scale head, would this be preferable to disabling normalisation in the training data?

Thank you for your time and help!

**Comment by Shexiaox** [comment]
Also, regarding the statement in the paper that "better results can be obtained by using the iterative refinement design of the original VGGT", if I want a more accurate camera pose, would it be a better option to modify it to resemble the original iterative camera head?

**Comment by jytime** [comment]
Hi @Shexiaox  

1. yes, loss_conf_depth
2. better to use them as token inputs before the aggregator
3. yes only after finetuning. it cannot be used in this way off the shelf.
4. even a simple frame index could work, but i believe there is a better solution
5. adding an additional head like camera head (probably you can call it scale head), and estimate the scale for the sequence or for the frames
6. yes always better, but needs finetuning

**Comment by Shexiaox** [comment]
> Hi [@Shexiaox](https://github.com/Shexiaox)
> 
> 1. yes, loss_conf_depth
> 2. better to use them as token inputs before the aggregator
> 3. yes only after finetuning. it cannot be used in this way off the shelf.
> 4. even a simple frame index could work, but i believe there is a better solution
> 5. adding an additional head like camera head (probably you can call it scale head), and estimate the scale for the sequence or for the frames
> 6. yes always better, but needs finetuning

Thank you very much for your detailed reply
